pilot/pkg/model/service.go

```diff
@@ -487,6 +492,38 @@ type ServiceAttributes struct {
 	// The port that the user provides in the meshNetworks config is the service port.
 	// We translate that to the appropriate node port here.
 	ClusterExternalPorts map[cluster.ID]map[uint32]uint32
+
+	// ExternalTrafficPolicy affects the list of available endpoints available in case
+	// of NodePort services. This is useful to preserve source IP address. If the traffic
+	// is sent to a node which does not have a workload then it would be dropped.
+	ExternalTrafficPolicy ExternalTrafficPolicy
+}
+
+type ExternalTrafficPolicy string
```

Comment:
It looks we donot need to redefine them again

Reply:
Removed it.